vision_evaluation/evaluators.py

```diff
@@ -1046,3 +1046,78 @@ def add_predictions(self, predictions, targets):
             gt_mask = np.asarray(gt_mask)
             mean_l1 = np.abs(pred_mask.astype(np.float)-gt_mask.astype(np.float)).mean()
             self._metric_sum += mean_l1
+
+
+class GroupWiseEvaluator(Evaluator):
+    """
+    Group wise evaluator: calculates metrics for the provided evaluator for each target class and group class separately.
+    """
+
+    def __init__(self, evaluator: Evaluator):
+        super().__init__()
+        self._evaluator = evaluator
+        self.store = None
+
+    def _get_id(self):
+        return "group_wise_accuracy"
```

Comment:
why is this restricted to accuracy? What if I am calculating group-wise average precision?